### PR TITLE
[Sophia]Tasks 21 to 30

### PR DESCRIPTION
**Pull request by SofiMat**

### CODE REVIEW

File: tasks/21_check-letters/check-letters.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+
+
+export function checkLetters(array) {
+  const str = array[0].toLowerCase();
+  const copy = array[1].toLowerCase().split('');
```
Comment (tarasbobak) [inline tasks/21_check-letters/check-letters.js]: string are like arrays, so you can access 'hello'[0] and you will get 'h'.

Comment (tarasbobak) [reply tasks/21_check-letters/check-letters.js]: no need for split

File: tasks/21_check-letters/check-letters.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+
```
Comment (tarasbobak) [inline tasks/21_check-letters/check-letters.js]: Good solution. But why there are 2 empty lines before implementation?

File: tasks/22_remove-falsy-values/remove-falsy-values.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+
```
Comment (tarasbobak) [inline tasks/22_remove-falsy-values/remove-falsy-values.js]: second file in a row?
is there a secret message on line 1?

**Comment by SofiMat** [comment]
I was removing eslint disable, they are all with that line

сб, 20 жовт. 2018 о 22:57 Taras <notifications@github.com> пише:

File: tasks/22_remove-falsy-values/remove-falsy-values.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
...
+export function removeFalsyValues(str) {
+  let result = str.filter((elems)=> !!elems);
```
Comment (tarasbobak) [inline tasks/22_remove-falsy-values/remove-falsy-values.js]: Questions:
1) why let?
2) why do you need result variable?
3) if there is only one argument why not simplify arrow function notation to
return str.filter(elem=> !!elem);

**Comment by SofiMat** [comment]
Should I change it now?

сб, 20 жовт. 2018 о 22:59 Sofiya Matviishyn <sofiyamatviishyn@gmail.com>
пише:

File: tasks/23_seek-destroy/seek-destroy.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+export function destroyer(arr) {
+  /* eslint-disable-next-line */
+  let args = Array.prototype.slice.call(arguments, 1);
```
Comment (tarasbobak) [inline tasks/23_seek-destroy/seek-destroy.js]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters

File: tasks/23_seek-destroy/seek-destroy.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
...
+  function des(smth) {
```
Comment (tarasbobak) [inline tasks/23_seek-destroy/seek-destroy.js]: smth? do you thing the reader of a code will understand what is smth?

File: tasks/24_index-to-insert/index-to-insert.js [hunk 7]
```diff
@@ -0,0 +1,13 @@
+
+export function getIndexToInsert(arr, val) {
+  arr.sort(function(a, b) {
```
Comment (tarasbobak) [inline tasks/24_index-to-insert/index-to-insert.js]: Hm, is array not sorted?)

File: tasks/24_index-to-insert/index-to-insert.js [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+    return a - b;
+  });
+  for (let a in arr) {
```
Comment (tarasbobak) [inline tasks/24_index-to-insert/index-to-insert.js]: https://stackoverflow.com/questions/3010840/loop-through-an-array-in-javascript

File: tasks/24_index-to-insert/index-to-insert.js (hunk 7 above)
Comment (SofiMat) [reply tasks/24_index-to-insert/index-to-insert.js]: my fault, didn't notice it

File: tasks/25_rot13-encoder/rot13-encoder.js [hunk 9]
```diff
@@ -0,0 +1,8 @@
+
+export function rot13Encoder(str) {
+  let input = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz';
+  let output = 'NOPQRSTUVWXYZABCDEFGHIJKLMnopqrstuvwxyzabcdefghijklm';
```
Comment (tarasbobak) [inline tasks/25_rot13-encoder/rot13-encoder.js]: Hardcoded solution?
This is smart. But will not be good for real life implementation.
Why let and not const?

File: tasks/26_average-mark/average-mark.js [hunk 10]
```diff
@@ -0,0 +1,8 @@
+export function getAverage(arr) {
+  let sum = 0;
+  for (let i = 0; i < arr.length; i++) {
+    sum += arr[i];
+  }
+  let avg = sum / arr.length;
+  return Math.floor(avg);
```
Comment (tarasbobak) [inline tasks/26_average-mark/average-mark.js]: Why Math.floor, but not let's say Math.round?

File: tasks/27_basic-subclasses/basic-subclasses.js [hunk 11]
```diff
@@ -0,0 +1,24 @@
+
```
Comment (tarasbobak) [inline tasks/27_basic-subclasses/basic-subclasses.js]: Hello, I'm empty line.
I don't know why I'm here...

File: tasks/28_vowel-count/vowel-count.js [hunk 12]
```diff
@@ -0,0 +1,12 @@
+
+export function getVowelCount(str) {
```
Comment (tarasbobak) [inline tasks/28_vowel-count/vowel-count.js]: Simple and good solution

File: tasks/29_remove-duplicates/remove-duplicates.js [hunk 13]
```diff
@@ -0,0 +1,7 @@
+
+export function removeDuplicates(duplicatesArray) {
+  let uniqueArray = duplicatesArray.filter(function(elem, pos) {
```
Comment (tarasbobak) [inline tasks/29_remove-duplicates/remove-duplicates.js]: Absolutely great solution.
But const, not let

File: tasks/29_remove-duplicates/remove-duplicates.js [hunk 14]
```diff
@@ -0,0 +1,7 @@
...
+    return duplicatesArray.indexOf(elem) == pos;
```
Comment (tarasbobak) [inline tasks/29_remove-duplicates/remove-duplicates.js]: Use strict comparison as default please ===

File: tasks/30_people-in-bus/people-in-bus.js [hunk 15]
```diff
@@ -0,0 +1,13 @@
+
+export function peopleInBus(busStops) {
+  let totalCount = 0;
+  let busStopsLeng = busStops.length;
+  for (let i=0; i< busStopsLeng; i++) {
```
Comment (tarasbobak) [inline tasks/30_people-in-bus/people-in-bus.js]: It is very very cumbersome. Can all of these
```
    let currentStop = busStops[i];
    let gettingOn = currentStop[0];
    let gettingOff = currentStop[1];
    totalCount += gettingOn;
    totalCount -= gettingOff;
```
be placed in one line?
Please at least try in console to implement it with reduce.

**Comment by tarasbobak** [comment]
@SofiMat Some of your solutions are just great. Some are not good.
They all pass tests, but it's not enough. I want you to learn on your mistakes.
When you finish your task spend 5 more minutes to have a side-view on your code. Fix indentation, remove redundant code. I know it's hard to do from the beginning.
You have to shift you thinking from writing for yourself to writing for anybody who would read you code next.
Pay more attention to naming. Think of better names for variables, commits, etc.
Spend 1 hour to learn about map, filter, reduce, forEach on MDN.
Do not use for in for arrays.

**Pull request merged**